Level 2

DenisIlyushin opened the pull request.

DenisIlyushin committed:
init

Базовая структура проекта

DenisIlyushin committed:
feat: сделать неавторизованный хедер

DenisIlyushin committed:
feat: сделать авторизованный хедер

DenisIlyushin committed:
feat: сделать авторизованный хедер

DenisIlyushin committed:
feat: начал footer

DenisIlyushin committed:
feat: сделать липкий футер

DenisIlyushin committed:
feat: доделать футер

DenisIlyushin committed:
fix: поправить ошибки футера

DenisIlyushin committed:
feat: начать landing, поправить семантику

DenisIlyushin committed:
feat: сделать AboutProject

DenisIlyushin committed:
feat: поправить боковые отступы макета

DenisIlyushin committed:
feat: сделать Techs

DenisIlyushin committed:
feat: сделать AboutMe

DenisIlyushin committed:
feat: подправить AboutMe

DenisIlyushin committed:
feat: исправить ошибку манифеста

DenisIlyushin committed:
feat: сделать Portfolio

DenisIlyushin committed:
feat: накостылять навигацию

DenisIlyushin committed:
feat: накостылять навигацию

DenisIlyushin committed:
feat: подправить адаптив AboutMe чтобы глаза не резало

DenisIlyushin committed:
feat: сделать бургер-меню

DenisIlyushin committed:
feat: поправить код-стиль

DenisIlyushin committed:
feat: поправить код-стиль

DenisIlyushin committed:
feat: сделать адаптив 768

DenisIlyushin committed:
feat: фикс ссылки в портфолио

DenisIlyushin committed:
feat: пересобрать бургер меню

DenisIlyushin committed:
feat: сделать лендинг 320

DenisIlyushin committed:
feat: сделать лендинг 320

DenisIlyushin committed:
feat: Поправить переходы между экранами на свой вкус

DenisIlyushin committed:
feat: поправить бургер-меню 320 и закончить адаптив лендинга

DenisIlyushin committed:
feat: добавить ProtectedRoute.jsx

DenisIlyushin committed:
feat: Начать работу над Movies

DenisIlyushin committed:
feat: исправить поведение Header при навигации по страницам

DenisIlyushin committed:
feat: поправить семантику в Movies

DenisIlyushin committed:
feat: сделать каркас поисковой строки

DenisIlyushin committed:
feat: странное наблюдение

DenisIlyushin committed:
feat: разгадать загадку

DenisIlyushin committed:
feat: разгадать загадку

DenisIlyushin committed:
feat: разгадать загадку

DenisIlyushin committed:
feat: поправить нейминг

DenisIlyushin committed:
feat: сделать каркас MoviesList

DenisIlyushin committed:
feat: сделать каркас MoviesList

DenisIlyushin committed:
feat: собрать компоненты Movies и SavedMovies

DenisIlyushin committed:
feat: доработать SearchForm.css

DenisIlyushin committed:
feat: добавить кнопку еще

DenisIlyushin committed:
feat: доработать адаптив SavedMovies и Movies

DenisIlyushin committed:
feat: Записать переходы экрана в приложении

DenisIlyushin committed:
feat: Переписать селекторы ToggleSwitch по БЭМ

DenisIlyushin committed:
feat: Поправить константы

DenisIlyushin committed:
feat: Поправить навигацию

DenisIlyushin committed:
feat: Добавить хук стандартной валидации форм

DenisIlyushin committed:
feat: Дабавить профиль и обработку

DenisIlyushin committed:
feat: Поправить хедер

DenisIlyushin committed:
feat: Поправить MoviesCard

DenisIlyushin committed:
feat: Убрать валидацию ошибок формы Profile

не было в задании

DenisIlyushin committed:
feat: Поправить поведение неактивной кнопки

DenisIlyushin committed:
feat: добавить заготовки для авторизации

DenisIlyushin committed:
feat: подготовить состояние для авторизации

DenisIlyushin committed:
feat: доработать адаптив Profile

DenisIlyushin committed:
feat: .gitignore

DenisIlyushin committed:
feat: .gitignore

DenisIlyushin committed:
feat: сделать авторизацию целиком

DenisIlyushin committed:
feat: 404 в процессе

DenisIlyushin committed:
feat: мелочи

DenisIlyushin committed:
feat: мелочи

DenisIlyushin committed:
feat: добавить Preloader

DenisIlyushin committed:
feat: добавить Preloader

DenisIlyushin committed:
feat: Убрать лишние переменные

AliaksandrHrychonak reviewed (commented):
# Резюме по работе:

## Верстка и JSX

### Критерии, влияющие на работоспособность

- [ ] <!--12--> Все блоки и элементы из макета сверстаны. Размеры и расположение не отличаются от макета на более, чем на 30px на десктопной версии и на 10px на остальных размерах экрана, для которых подготовлен макет. Отличия в размерах не должны визуально противоречить макету и выбиваться из заложенной в макет сетки, в этом случае требуется более строгое соответствие размерам макета.
    - приложение следует перестраивать в мобильный режим на точке 768px, в шапке достаточно места для отображения десктопной навигации
    - **На 1280px не по макету:**
    - https://disk.yandex.com/d/dvOynlF0r3bJOg
    - **На 768px не по макету:**
    - https://disk.yandex.com/d/mS3Y4LO5KEhVbw
    - https://disk.yandex.com/d/CLSjKG5qg99sSA
    - https://disk.yandex.com/d/mzdY-DxfyOInXg
    - https://disk.yandex.com/d/evdaGwEdhNmIFQ
    - https://disk.yandex.com/d/u3TsAA9YpUAQ0Q
    - https://disk.yandex.com/d/scCwbvwekrYPDw
    - **На 320px не по макету:**
    - https://disk.yandex.com/d/wlcu1qDRk7B6Cw
    - https://disk.yandex.com/d/rL-MZS2U1doYfA
    - https://disk.yandex.com/d/lDC8NzpeRI55Cg
    - https://disk.yandex.com/d/OCim63ID-oyxaQ
    - https://disk.yandex.com/d/icjGEXfC-sCLSA
    - https://disk.yandex.com/d/LyxKauo5fxdnYA
    
    >Для более точного контроля отображения вёрстки можно воспользоваться плагином `PerfectPixel` для `Chrome`, можно в фигме выделить весь >макет, экспортировать его картинкой и наложить на вёрстку в браузере. Плагин позволяет наложить изображение на вёрстку в браузере, >тем самым можно слегкостью подправить недочёты по отображению.

- [x] <!--10.55--> Проект адаптирован под различные разрешения экрана. Горизонтальный скролл не возникает на разрешениях от 320 пикселей и больше. Нельзя скрывать полосу прокрутки свойством overflow: hidden.
- [x] <!--9.55--> Корректно работает навигация между страницами и ссылки на внешние ресурсы: ни одна ссылка не ведёт в пустоту или на якорь, внешние ссылки открываются в новой вкладке. На все страницы проекта можно попасть, а скрытые блоки можно отобразить.
- [ ] <!--5.55--> Именование классов сделаны по БЭМ.
    - проект имеет БЭМ ошибки, удобно проверить валидатором https://nglazov.github.io/bem-validator-page/ , проконтролируйте данный критерий по всем роутам, пример https://disk.yandex.com/i/NX_G2mn0wds2iQ
- [x] <!--8.55--> Отзывчивая вёрстка, которая корректно тянется на всех промежуточных разрешениях.
- [ ] <!--3.38--> В коде используется семантическая разметка: применяются семантические теги, выбор элементов при вёрстке корректен (параграф должен быть параграфом, список — списком); структура DOM-дерева состоит не только из контейнеров div.
    - проект имеет ошибки валидации, удобно проверить валидатором https://validator.w3.org/#validate_by_input , проконтролируйте данный критерий по всем роутам, пример https://disk.yandex.com/i/V97aN-A6B4zXpg
    - Контентное содержимое страницы нужно обернуть тегом `<main>` **по всем роутам** (всё кроме `header и  footer`)
    - в роуте `/главная, /signin , /signup, /404, /profile, /movies, /saved-movies` не хватает сематического тега `<main>`
    - Поисковая форма является отдельной секцией
    - `nav-tab__item movies-card-list__content` однотипные элементы следует группировать с помощью списка `<ul>`
    - `header__auth_navigation(для `header__auth_button`)` элементы представляют навигацию `<nav>`
    - `profile__title` является заголовком первого уровня
    - Всем кнопкам в проекте нужен заполненный атрибут `type`, если кнопка **не** отвечает за отправку формы то значение будет `"button"`
- [x] <!--3.38--> Для позиционирования элементов выбран верный подход, описанный корректным синтаксисом.
- [x] <!--3.38--> Каркас макета реализован на Flex layout или Grid layout;
- [x] <!--3.38--> Инфраструктурные файлы проекта созданы через CRA.
- [x] <!--3.38--> Работа модальных окон настроена.
- [x] Разметка портирована в JSX:
    - [x] <!--3.38--> разметка заключена в (), лишние пустые <div> для этого не используются;
    - [x] <!--3.38--> разметка вынесена в соответствующие ей компоненты.
- [x] В проекте есть:
    - [x] <!--3.38--> директория images/ с картинками,
    - [x] <!--3.38--> директория components/ с JS и CSS-файлами компонентов,
    - [x] <!--3.38--> директория fonts/ со шрифтами.
    
### Хорошие практики

- [ ] <!--2.14--> Кнопки, инпуты и ссылки реализованы во всех состояниях, указанных в макете.
    - Всем интерактивным элементам(ссылки, кнопки, чекбоксы, лого) нужен эффект при наведении и выставленный курсор, проконтролируйте данный критерий для всех интерактивных элементов в проекте
        - например не хватает для `(лого по всем роутам), чекбокс` 
- [x] <!--2.14--> В макете есть одинаковые элементы. Для их оформления должен быть переиспользован один компонент.
- [x] <!--2.14--> Шрифты подключены через @font-face.
- [x] <!--2.14--> Иконки экспортированы в формате SVG.
- [ ] <!--2.14--> Элементы формы должны выделяться, когда на них установлен фокус. Состояние фокуса, а также outline стилизованы в соответствии с макетом или на своё усмотрение.
    - Всем инпутам проекта нужен стилизованный фокус и переопределённый `outline`
- [ ] <!--2.14--> У формы должны быть заданы плейсхолдеры, валидированы обязательные поля.
    - Всем инпутам проекта нужно добавить плейсхолдеры
    - Всем полям **Имя и Пароль** нужна валидация минимальной/максимальной длинны значения
- [x] <!--2.14--> reset.css запрещен.

### Рекомендации

- [x] <!--1.68--> Для каждого шрифта указаны альтернативные варианты из системных шрифтов.
- [x] <!--1.67--> Для изображений задан атрибут alt с подходящий значением.
- [x] <!--1.67--> Растровые и векторные изображения оптимизированы;
    
## Количество баллов: **73**

DenisIlyushin committed:
rev1: исправить верстку Profile.css

DenisIlyushin committed:
rev1: исправить верстку Landing 768px

DenisIlyushin committed:
rev1: исправить верстку Movies и SavedMovies 768px

DenisIlyushin committed:
rev1: исправить верстку Movies и SavedMovies 320px

DenisIlyushin committed:
rev1: исправить верстку Landing 320px

DenisIlyushin committed:
rev1: исправить замечания БЭМ

проверено валидатором https://nglazov.github.io/bem-validator-page/

DenisIlyushin committed:
rev1: собрать изображения для pixelPerfect

DenisIlyushin committed:
rev1: поправить семантику верстки

Контентное содержимое страницы нужно обернуть тегом <main> по всем роутам (всё кроме header и  footer)
в роуте /главная, /signin , /signup, /404, /profile, /movies, /saved-movies не хватает сематического тега <main>
Поисковая форма является отдельной секцией
nav-tab__item movies-card-list__content однотипные элементы следует группировать с помощью списка <ul>
header__auth_navigation(для header__auth_button) элементы представляют навигацию <nav>
profile__title является заголовком первого уровня
Всем кнопкам в проекте нужен заполненный атрибут type, если кнопка не отвечает за отправку формы то значение будет "button"

DenisIlyushin committed:
rev1: поправить интерактив при наведении на кнопки

DenisIlyushin committed:
rev1: переопределить outline для focused состояния инпутов

DenisIlyushin committed:
rev1: добавить проверку длины имени и пароля для всех полей, плейсхолдеры и поведение кнопок сабмита авторизации.

DenisIlyushin committed:
rev1: убрать лишнюю константу

DenisIlyushin committed:
rev1: проправить БЭМ еще раз

DenisIlyushin committed:
rev1: вроде бы все...

AliaksandrHrychonak reviewed (commented):
# Резюме по работе:

## Верстка и JSX

### Критерии, влияющие на работоспособность

- [ ] <!--12--> Все блоки и элементы из макета сверстаны. Размеры и расположение не отличаются от макета на более, чем на 30px на десктопной версии и на 10px на остальных размерах экрана, для которых подготовлен макет. Отличия в размерах не должны визуально противоречить макету и выбиваться из заложенной в макет сетки, в этом случае требуется более строгое соответствие размерам макета.
    - приложение следует перестраивать в мобильный режим на точке 768px, в шапке достаточно места для отображения десктопной навигации https://disk.yandex.com/i/aksHx9EUsd4UHA , до точки 768px кнопка мобильного меню не должна появляться в проекте
    - **1280**
    - https://disk.yandex.com/i/AY5AM_292wuO1w
    - **768**
    - https://disk.yandex.com/i/3_ejVGIoKlNqeQ
    - **320**
    - https://disk.yandex.com/i/9qBWgy_YY5DGEw
    - https://disk.yandex.com/i/A2fNzZeZqo9eFA
- [x] <!--10.55--> Проект адаптирован под различные разрешения экрана. Горизонтальный скролл не возникает на разрешениях от 320 пикселей и больше. Нельзя скрывать полосу прокрутки свойством overflow: hidden.
- [x] <!--9.55--> Корректно работает навигация между страницами и ссылки на внешние ресурсы: ни одна ссылка не ведёт в пустоту или на якорь, внешние ссылки открываются в новой вкладке. На все страницы проекта можно попасть, а скрытые блоки можно отобразить.
- [ ] <!--5.55--> Именование классов сделаны по БЭМ.
    - проект имеет БЭМ ошибки, удобно проверить валидатором https://nglazov.github.io/bem-validator-page/ , проконтролируйте данный критерий по всем роутам, пример https://disk.yandex.com/i/kwXKP3I-6qs8oQ
- [x] <!--8.55--> Отзывчивая вёрстка, которая корректно тянется на всех промежуточных разрешениях.
- [x] <!--3.38--> В коде используется семантическая разметка: применяются семантические теги, выбор элементов при вёрстке корректен (параграф должен быть параграфом, список — списком); структура DOM-дерева состоит не только из контейнеров div.
- [x] <!--3.38--> Для позиционирования элементов выбран верный подход, описанный корректным синтаксисом.
- [x] <!--3.38--> Каркас макета реализован на Flex layout или Grid layout;
- [x] <!--3.38--> Инфраструктурные файлы проекта созданы через CRA.
- [x] <!--3.38--> Работа модальных окон настроена.
- [x] Разметка портирована в JSX:
    - [x] <!--3.38--> разметка заключена в (), лишние пустые <div> для этого не используются;
    - [x] <!--3.38--> разметка вынесена в соответствующие ей компоненты.
- [x] В проекте есть:
    - [x] <!--3.38--> директория images/ с картинками,
    - [x] <!--3.38--> директория components/ с JS и CSS-файлами компонентов,
    - [x] <!--3.38--> директория fonts/ со шрифтами.
    
### Хорошие практики

- [x] <!--2.14--> Кнопки, инпуты и ссылки реализованы во всех состояниях, указанных в макете.
- [x] <!--2.14--> В макете есть одинаковые элементы. Для их оформления должен быть переиспользован один компонент.
- [x] <!--2.14--> Шрифты подключены через @font-face.
- [x] <!--2.14--> Иконки экспортированы в формате SVG.
- [ ] <!--2.14--> Элементы формы должны выделяться, когда на них установлен фокус. Состояние фокуса, а также outline стилизованы в соответствии с макетом или на своё усмотрение.
    - Поисковому инпуту нужен стилизованный фокус и переопределённый `outline`
- [x] <!--2.14--> У формы должны быть заданы плейсхолдеры, валидированы обязательные поля.
- [x] <!--2.14--> reset.css запрещен.

### Рекомендации

- [x] <!--1.68--> Для каждого шрифта указаны альтернативные варианты из системных шрифтов.
- [x] <!--1.67--> Для изображений задан атрибут alt с подходящий значением.
- [x] <!--1.67--> Растровые и векторные изображения оптимизированы;
    
## Количество баллов: **80**

DenisIlyushin committed:
rev2: исправить SearchForm

Поисковому инпуту нужен стилизованный фокус и переопределённый outline

DenisIlyushin committed:
rev2: исправить бэм в NavTab

проект имеет БЭМ ошибки, удобно проверить валидатором https://nglazov.github.io/bem-validator-page/ , проконтролируйте данный критерий по всем роутам, пример https://disk.yandex.com/i/kwXKP3I-6qs8oQ

DenisIlyushin committed:
rev2: исправить ошибки верстки 1280

1280
https://disk.yandex.com/i/AY5AM_292wuO1w

DenisIlyushin commented:
404 ошибка мне больше нравится посередине страницы для tablet и mobile. Раз работу приняли, я оставлю так )
